pkg/api/v2/remove.go

```diff
@@ -11,11 +11,20 @@ import (
 // RemoveFromDqliteRequest represents a request to remove a node from the dqlite cluster.
 type RemoveFromDqliteRequest struct {
 	// RemoveEndpoint is the endpoint of the node to remove from the dqlite cluster.
-	RemoveEndpoint string `json:"removeEndpoint"`
+	RemoveEndpoint string `json:"remove_endpoint"`
 }
 
 // RemoveFromDqlite implements the "POST /v2/dqlite/remove" endpoint and removes a node from the dqlite cluster.
-func (a *API) RemoveFromDqlite(ctx context.Context, req RemoveFromDqliteRequest) (int, error) {
+func (a *API) RemoveFromDqlite(ctx context.Context, req RemoveFromDqliteRequest, token string) (int, error) {
+	isValid, err := a.Snap.IsCAPIAuthTokenValid(token)
```

Comment:
I really like what you're saying, but IIUC the middlewares are not configurable per endpoint in cluster-agent: https://github.com/canonical/microk8s-cluster-agent/blob/main/pkg/server/server.go#L45-L46
All the v1 or v2 endpoints get registered with the same middleware. 
We technically can have this `capi-auth-token` header check as a middleware, but since it's going to get applied for the other v2 endpoints as well (`/image/import` and `/join`) we need to make it optional, which is counter intuitive. WDYT?

Reply:
You can simply set a new middleware on a per route basis:
https://github.com/canonical/microk8s-cluster-agent/blob/main/pkg/server/server.go#L19-L27

For this route you would add it here: 
https://github.com/canonical/microk8s-cluster-agent/pull/56/files#diff-6ecd338993def4d5349db1e632bfa290c6aa5d6a7e04e6413f0f27e8983161f3R58

Reply:
My bad, thanks for pointing this out. Will do.

Reply:
I'm reverting this. Not really happy with the result...

Reply:
ok, up to you. This was only a suggestion. If you think the current implementation is more straight-forward - go for it :)